tools/installer: create shareable installer components

lahm86 opened the pull request:
Part of #2681.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

This is working towards a TR2 installer, and initially dismantles the TR1 installer to make shareable components ready. As part of this, all of the text is now external to make the UI easier to feed game specific details to. I'll build on this PR incrementally.

I've also carried out some maintenance to remove VS warnings and messages. I was toying with updating to .NET 8 as 6 is out of support, but that's probably best for another time.

The TR1 installer as part of this should behave as it does currently, bar some very minor WPF updates.

lahm86 committed:
tr2/data: create images directory

This updates the game flows to look in a dedicated images directory in
order to tidy up the main data folder. Default images provided as PNG
to avoid users having to copy manually.

lahm86 committed:
tools/installer: create common installer library

This creates a generic common installer WPF library for both games.

lahm86 committed:
tools/installer: migrate TR1X installer

This migrates the TR1X installer to use the new common library.

lahm86 committed:
tools/installer: move language to external file

This moves all language used in the installer to an embedded JSON file
to allow for customisation per game.

lahm86 committed:
tools/installer: hide expansion options unless enabled

lahm86 committed:
tools/installer: decouple shortcut and expansion logic

This decouples the shortcut creation so it can be used generically in
either game. It also allows for the expansion pack zip names to be
defined externally rather than guessing from the selected type.

lahm86 committed:
tools/installer: rename solution

lahm86 committed:
tools/installer: standardize output paths

This standardizes output paths to conform with shipped TRX data.
Folders and files created within the target folder will be in
lower-case; PCX files will be copied to a sub-directory within images
as TRX ships with custom images, and this as a result keeps the data
folder tidy.

lahm86 commented:
Pushed a change for the OG `.pcx` shift as discussed, LMK if this looks OK. Also standardized output paths so regardless of source, it should match the structure in the release zip (e.g. `data` rather than `DATA`). I also decoupled the UB download logic as I'd missed this before, so it's in turn ready for TR2. Worth another check, thanks!

aredfan commented:
The TR1X directory looks much nicer and orderly. The og images are in the right place too, thank you and much appreciated. 👍 

I can confirm all the files are present after doing a fresh install, LGTM.

lahm86 committed:
tools/installer: create TR2 installer

This creates a TR2 installer in similar fashion to TR1. The installers
themselves are basically directory tree copiers, rather than having to
extract from BIN/ISO like TR1.

rr- reviewed (commented):
The new directory structure probably needs reflecting in the documentation, right?

lahm86 commented:
> The new directory structure probably needs reflecting in the documentation, right?

Thanks, good point. I haven't done a changelog entry either. I'll wrap up the docs, adding similar install details as TR1 in the readme etc.

lahm86 committed:
docs: update docs

aredfan commented on docs/tr2/README.md:
```diff
@@ -9,14 +9,135 @@ game with new enhancements and features.
+│   ├── level5.tr2
+│   ├── living.tr2
+│   ├── main.sfx
+│   ├── monastry.tr2
```
I noticed the `main_gm.sfx` file is missing here.

The rest of the documentation looks good.

lahm86 commented:
Last push fixes some whitespace issues only in C# code.
I think we're pretty much good to go, just need to organise the resource uploads and automated installer build for TR2.

rr- committed:
build: add TR2 installer automatic builds
